src/gui/FileBrowser.cpp

```diff
@@ -556,36 +672,29 @@ void FileBrowserTreeWidget::mouseReleaseEvent(QMouseEvent * me )
 {
 	m_mousePressed = false;
 
-	m_pphMutex.lock();
-	if( m_previewPlayHandle != NULL )
+	QMutexLocker previewLocker(&m_pphMutex);
+
+	if (m_previewPlayHandle != nullptr)
 	{
-		// if there're samples shorter than 3 seconds, we don't
+		// If less than 3 seconds remain of the sample, we don't
 		// stop them if the user releases mouse-button...
-		if( m_previewPlayHandle->type() == PlayHandle::TypeSamplePlayHandle )
+		if (m_previewPlayHandle->type() == PlayHandle::TypeSamplePlayHandle)
 		{
-			SamplePlayHandle * s = dynamic_cast<SamplePlayHandle *>(
-							m_previewPlayHandle );
-			if( s && s->totalFrames() - s->framesDone() <=
-				static_cast<f_cnt_t>( Engine::mixer()->
-						processingSampleRate() * 3 ) )
+			SamplePlayHandle* s = dynamic_cast<SamplePlayHandle*>(m_previewPlayHandle);
+			auto second = static_cast<f_cnt_t>(Engine::mixer()->processingSampleRate());
+			if (s && s->totalFrames() - s->framesDone() <= second * 3)
 			{
-				s->setDoneMayReturnTrue( true );
-				m_previewPlayHandle = NULL;
-				m_pphMutex.unlock();
-				return;
+				s->setDoneMayReturnTrue(true);
 			}
+			else { stopPreview(); }
 		}
-		Engine::mixer()->removePlayHandle( m_previewPlayHandle );
-		m_previewPlayHandle = NULL;
 	}
```

Comment:
This now stops sample previews with more than three seconds left correctly, but preset previews now go on forever.

Reply:
Good catch, that was poor testing on my part! Fixed and tested, works properly again for me.